src/Microsoft.Health.Dicom.SqlServer/Features/Store/SqlIndexDataStore.cs

```diff
@@ -5,284 +5,99 @@
-                VLatest.RetrieveDeletedInstance.PopulateCommand(
-                    sqlCommandWrapper,
-                    batchSize,
-                    maxRetries);
-
-                using (var reader = await sqlCommandWrapper.ExecuteReaderAsync(CommandBehavior.SequentialAccess, cancellationToken))
-                {
-                    try
-                    {
-                        while (await reader.ReadAsync(cancellationToken))
-                        {
-                            (string studyInstanceUid, string seriesInstanceUid, string sopInstanceUid, long watermark) = reader.ReadRow(
-                                VLatest.DeletedInstance.StudyInstanceUid,
-                                VLatest.DeletedInstance.SeriesInstanceUid,
-                                VLatest.DeletedInstance.SopInstanceUid,
-                                VLatest.DeletedInstance.Watermark);
-
-                            results.Add(new VersionedInstanceIdentifier(
-                                studyInstanceUid,
-                                seriesInstanceUid,
-                                sopInstanceUid,
-                                watermark));
-                        }
-                    }
-                    catch (SqlException ex)
-                    {
-                        throw new DataStoreException(ex);
-                    }
-                }
-            }
-
-            return results;
+            await InitInnerDataStoreAsync();
+            return await _innerDataStore.RetrieveDeletedInstancesAsync(batchSize, maxRetries, cancellationToken);
         }
 
-        public async Task DeleteDeletedInstanceAsync(VersionedInstanceIdentifier versionedInstanceIdentifier, CancellationToken cancellationToken = default)
+        public async Task UpdateInstanceIndexStatusAsync(VersionedInstanceIdentifier versionedInstanceIdentifier, IndexStatus status, CancellationToken cancellationToken)
         {
-            await _sqlServerIndexSchema.EnsureInitialized();
-
-            using (SqlConnectionWrapper sqlConnectionWrapper = await _sqlConnectionFactoryWrapper.ObtainSqlConnectionWrapperAsync(cancellationToken, true))
-            using (SqlCommandWrapper sqlCommandWrapper = sqlConnectionWrapper.CreateSqlCommand())
-            {
-                VLatest.DeleteDeletedInstance.PopulateCommand(
-                    sqlCommandWrapper,
-                    versionedInstanceIdentifier.StudyInstanceUid,
-                    versionedInstanceIdentifier.SeriesInstanceUid,
-                    versionedInstanceIdentifier.SopInstanceUid,
-                    versionedInstanceIdentifier.Version);
-
-                try
-                {
-                    await sqlCommandWrapper.ExecuteScalarAsync(cancellationToken);
-                }
-                catch (SqlException ex)
-                {
-                    throw new DataStoreException(ex);
-                }
-            }
+            await InitInnerDataStoreAsync();
```

Comment:
can this be done in a constructor instead of each method

Reply:
Would this to be too much heavy to put in constructor?